uvicorn/lifespan/on.py

```diff
@@ -58,13 +60,25 @@ def __init__(self, config):
     async def send(self, message):
         assert message["type"] in (
             "lifespan.startup.complete",
+            "lifespan.startup.failed",
             "lifespan.shutdown.complete",
         )
 
         if message["type"] == "lifespan.startup.complete":
             assert not self.startup_event.is_set(), STATE_TRANSITION_ERROR
             assert not self.shutdown_event.is_set(), STATE_TRANSITION_ERROR
             self.startup_event.set()
+
+        elif message["type"] == "lifespan.startup.failed":
+            assert not self.startup_event.is_set(), STATE_TRANSITION_ERROR
+            assert not self.shutdown_event.is_set(), STATE_TRANSITION_ERROR
+            self.startup_event.set()
+
+            if message.get("message", ""):
```

Comment:
I guess this could just be `message.get("message")`, since it'll return `None` off not present.

Reply:
Fixed!